dotnetv3/S3/LifecycleExample/LifecycleExample/Lifecycle.cs

```diff
@@ -0,0 +1,202 @@
+                                new LifecycleTransition
+                                {
+                                    Days = 30,
+                                    StorageClass = S3StorageClass.StandardInfrequentAccess,
+                                },
+                                new LifecycleTransition
+                                {
+                                    Days = 365,
+                                    StorageClass = S3StorageClass.Glacier,
+                                },
+                            },
+                            Expiration = new LifecycleRuleExpiration()
+                            {
+                                Days = 3650,
+                            },
+                        },
+                    },
+                };
+
+                // Add the configuration to the bucket.
+                await AddExampleLifecycleConfigAsync(client, bucketName, lifeCycleConfiguration);
+
+                // Retrieve the existing configuration to show that the configuration
+                // was added.
+                lifeCycleConfiguration = await RetrieveLifecycleConfigAsync(client, bucketName);
+
+                // Add another rule to the existing lifecycle rule.
+                lifeCycleConfiguration.Rules.Add(new LifecycleRule
+                {
+                    Id = "NewRule",
+                    Filter = new LifecycleFilter()
+                    {
+                        LifecycleFilterPredicate = new LifecyclePrefixPredicate()
+                        {
+                            Prefix = "YearlyDocuments/",
+                        },
+                    },
+                    Expiration = new LifecycleRuleExpiration()
+                    {
+                        Days = 3650,
+                    },
+                });
+
+                // Add the lifecycle configuration to the bucket.
+                await AddExampleLifecycleConfigAsync(client, bucketName, lifeCycleConfiguration);
+
+                // Verify that the the bucket now has three rules in its lifecycle configuration.
+                lifeCycleConfiguration = await RetrieveLifecycleConfigAsync(client, bucketName);
+                Console.WriteLine($"Expected # of rulest: 3; # of rules found: {lifeCycleConfiguration.Rules.Count}");
+
+                // Now delete the configuration from the bucket.
+                await RemoveLifecycleConfigAsync(client, bucketName);
+
+                // Show that the lifecycle configuration is no longer associated with
+                // the bucket.
+                lifeCycleConfiguration = await RetrieveLifecycleConfigAsync(client, bucketName);
+            }
+            catch (AmazonS3Exception ex)
+            {
+                Console.WriteLine($"Error encountered ***. Message:'{ex.Message}' when writing an object.");
+            }
+        }
+
+        /// <summary>
+        /// Adds lifecycle configuration information to the S3 bucket named in
```

Comment:
the S3 bucket -> from the Amzon S3 bucket

Reply:
No, S3 bucket is right.